extensions/gdx-tools/src/com/badlogic/gdx/tools/particleeditor/ParticleEditor.java

```diff
@@ -205,7 +205,11 @@ public void setVisible (String name, boolean visible) {
 	}
 
 	public ParticleEmitter getEmitter () {
-		return effect.getEmitters().get(effectPanel.editIndex);
+		Array<ParticleEmitter> emitters = effect.getEmitters();
+		if (effectPanel.editIndex < emitters.size) {
```

Comment:
This isn't as bulletproof as it maybe could be, since editIndex can sometimes be -1. It's probably fine, though, since that would have thrown an exception before, too.

Reply:
In theory the if condition here should be always true because when `emitterTable.getSelectedRow()` returns `-1` the `editIndex` field didn't get updated anymore. I left the check anyway because maybe there could be cases I didn't catch and prevent possible freezes.

Reply:
Yeah, in any case it's more robust than before.